pkg/syncer/specsyncer.go

```diff
@@ -86,20 +87,40 @@ func (c *Controller) deleteFromDownstream(ctx context.Context, gvr schema.GroupV
 	return c.toClient.Resource(gvr).Namespace(namespace).Delete(ctx, name, metav1.DeleteOptions{})
 }
 
+const namespaceLocatorAnnotation = "kcp.dev/namespace-locator"
+
 // TODO: This function is there as a quick and dirty implementation of namespace creation.
 //       In fact We should also be getting notifications about namespaces created upstream and be creating downstream equivalents.
-func (c *Controller) ensureNamespaceExists(ctx context.Context, namespace string) error {
+func (c *Controller) ensureNamespaceExists(ctx context.Context, namespace string, obj *unstructured.Unstructured) error {
 	namespaces := c.toClient.Resource(schema.GroupVersionResource{
 		Group:    "",
 		Version:  "v1",
 		Resource: "namespaces",
 	})
 
-	// Not found - try to create
 	newNamespace := &unstructured.Unstructured{}
 	newNamespace.SetAPIVersion("v1")
 	newNamespace.SetKind("Namespace")
 	newNamespace.SetName(namespace)
+
+	l := namespaceLocator{
+		LogicalCluster: obj.GetClusterName(),
+		Namespace:      obj.GetNamespace(),
+	}
+	b, err := json.Marshal(l)
+	if err != nil {
+		return err
+	}
+	newNamespace.SetAnnotations(map[string]string{
+		namespaceLocatorAnnotation: string(b),
+	})
+
+	if obj.GetLabels() != nil {
+		newNamespace.SetLabels(map[string]string{
```

Comment:
`// TODO: this should be set once at syncer startup and propagated around everywhere.`